core/templates/pages/splash-page/splash-page.component.ts

```diff
@@ -61,16 +64,78 @@ angular.module('oppia').component('splashPage', {
         return false;
       };
 
+      ctrl.onClickLearnMoreStudentsButton = function() {
+        $timeout(function() {
+          WindowRef.nativeWindow.location = '/learn/math';
+        }, 150);
+        return false;
+      };
+
+      ctrl.onClickLearnMoreTeachersButton = function() {
+        $timeout(function() {
+          WindowRef.nativeWindow.location = '/teach';
+        }, 150);
+        return false;
+      };
+
+      ctrl.getMainBackgroundUrl = function() {
+        if (ctrl.isWindowNarrow()) {
+          return ctrl.getStaticImageUrl('/splash/splashMainMobile.png');
+        } else {
+          return ctrl.getStaticImageUrl('/splash/splashMainDesktop.png');
+        }
+      };
+
+      ctrl.isWindowNarrow = function() {
+        return WindowDimensionsService.isWindowNarrow();
+      };
+
+      ctrl.onClickStartContributingButton = function() {
+        $timeout(function() {
+          WindowRef.nativeWindow.location = (
+            'https://www.oppiafoundation.org/volunteer');
+        }, 150);
+        return false;
+      };
+
       ctrl.onClickCreateExplorationButton = function() {
         SiteAnalyticsService.registerClickCreateExplorationButtonEvent();
         $timeout(function() {
           WindowRef.nativeWindow.location = '/creator-dashboard?mode=create';
         }, 150);
         return false;
       };
+
+      ctrl.incrementTestimonialId = function() {
+        ctrl.testimonialId = (ctrl.testimonialId + 1) % 3;
+      };
+
+      ctrl.decrementTestimonialId = function() {
+        ctrl.testimonialId = (((ctrl.testimonialId - 1) % 3) + 3) % 3;
+      };
+
       ctrl.$onInit = function() {
         ctrl.userIsLoggedIn = null;
+        ctrl.testimonialId = 0;
         LoaderService.showLoadingScreen('Loading');
+        var communityBackgroundImageName = 'dsk_community_background.png';
+        var testimonialBackgroundImageName = 'dsk_testimonial_background.png';
+        if (ctrl.isWindowNarrow()) {
+          communityBackgroundImageName = 'm_community_background.png';
+          testimonialBackgroundImageName = 'm_testimonial_background.png';
+        }
+        Array.from(
```

Comment:
This seems pretty fragile. Could we not do it this way? Instead, maybe use CSS or ng-class or something similar?

Reply:
Done, used ng-style.